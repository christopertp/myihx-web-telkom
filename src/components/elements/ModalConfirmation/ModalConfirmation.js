import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';
import Modal from '../Modal';
import styles from './styles.scoped.css';

export default function ModalConfirmation(props) {
  const { onClose, open, onAccept, onRefuse, title, subtitle, reverseBtn } = props;
  return (
    <Modal className={styles.root} onClose={onClose} open={open}>
      <h1>{title}</h1>
      <p>{subtitle}</p>
      {reverseBtn ? (
        <footer>
          <Button onClick={onRefuse} variant="bordered">
            No
          </Button>
          <Button onClick={onAccept}>Yes</Button>
        </footer>
      ) : (
        <footer>
          <Button onClick={onAccept} variant="bordered">
            Yes
          </Button>
          <Button onClick={onRefuse}>No</Button>
        </footer>
      )}
    </Modal>
  );
}

ModalConfirmation.defaultProps = {
  onAccept: () => {},
  onClose: () => {},
  onRefuse: () => {},
  open: false,
  reverseBtn: false,
  subtitle: '',
  title: '',
};

ModalConfirmation.propTypes = {
  onAccept: PropTypes.func,
  onClose: PropTypes.func,
  onRefuse: PropTypes.func,
  open: PropTypes.bool,
  reverseBtn: PropTypes.bool,
  subtitle: PropTypes.string,
  title: PropTypes.string,
};
