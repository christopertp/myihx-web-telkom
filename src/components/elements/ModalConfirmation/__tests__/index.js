import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ModalConfirmation from '../ModalConfirmation';

describe('src/components/elements/ModalConfirmation', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ModalConfirmation />);
    expect(tree).toMatchSnapshot();
    ModalConfirmation.defaultProps.onClose();
    ModalConfirmation.defaultProps.onAccept();
    ModalConfirmation.defaultProps.onRefuse();
  });

  test('reverse button', () => {
    const props = {
      ...ModalConfirmation.defaultProps,
      reverseBtn: true,
    };
    const result = ModalConfirmation(props);
    const { children } = result.props;
    expect(children[2].props.children[0].props.variant).toBe('bordered');
    expect(children[2].props.children[0].props.children).toBe('No');
    expect(children[2].props.children[1].props.children).toBe('Yes');
  });
});
