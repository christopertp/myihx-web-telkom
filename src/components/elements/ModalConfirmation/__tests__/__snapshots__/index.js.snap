// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/ModalConfirmation render 1`] = `
<Modal
  className=""
  onClose={[Function]}
  open={false}
>
  <h1>
    
  </h1>
  <p>
    
  </p>
  <footer>
    <Button
      className=""
      disabled={false}
      onClick={[Function]}
      to=""
      type="button"
      variant="bordered"
    >
      Yes
    </Button>
    <Button
      className=""
      disabled={false}
      onClick={[Function]}
      to=""
      type="button"
      variant="red"
    >
      No
    </Button>
  </footer>
</Modal>
`;
